client/modules/workspace/src/AppsBreadcrumb/AppsBreadcrumb.tsx

```diff
@@ -0,0 +1,61 @@
+import { Breadcrumb, BreadcrumbProps } from 'antd';
+import React, { useEffect, useState } from 'react';
+import { Link, useParams, useLocation } from 'react-router-dom';
+import styles from './AppsBreadcrumb.module.css';
+import { TAppParamsType, TAppResponse, useGetApps } from '@client/hooks';
+
+function getPathRoutes(path: string = '') {
+  const pathComponents = decodeURIComponent(path)
+    .split('/')
+    .filter((p) => !!p);
+
+  return pathComponents.map((comp, i) => ({
+    title: comp,
+    path: '/' + encodeURIComponent(pathComponents.slice(0, i + 1).join('/')),
+  }));
+}
+
+export const AppsBreadcrumb: React.FC<{
+  initialBreadcrumbs: { title: string; path: string }[];
+  path: string;
+} & BreadcrumbProps> = ({ initialBreadcrumbs, path, ...props }) => {
+  const [appData, setAppData] = useState<TAppResponse | null>(null);
+  const { appId } = useParams() as TAppParamsType;
+  const appVersion = new URLSearchParams(useLocation().search).get(
+    'appVersion'
+  ) as string | undefined;
+
+  const { data, isLoading } = useGetApps({ appId, appVersion });
+
+  useEffect(() => {
+    if (data) {
+      setAppData(data);
+    }
+  }, [data]);
+
+  const breadcrumbItems = [...initialBreadcrumbs, ...getPathRoutes(path)];
+
+  return (
+    <div className={styles.breadcrumbWrapper}>
+      {isLoading && <div>Loading...</div>}
+      {!isLoading && appData && (
```

Comment:
We want to always render the breadcrumb, i.e. Home / Use DesignSafe / Tools & Applications will always show here. We can render the app label if we detect an app is loaded

Reply:
I removed "appData &&" so that the breadcrumb will always render even when an app is not selected